### PR TITLE
Adding build file to sample-app-with-shared-lib to build it as Spring Boot app

### PR DESCRIPTION
**Pull request by jessiejuachon**

### CODE REVIEW

**Comment by jessiejuachon** [comment]
Hi @surite . We should add this build in the CI later. If you add it now, it will fail because of issues related to https://github.com/vmware/singleton/issues/882.
Regardless of the status of the issues, this PR can be merged so that  developers and testers can already use it to test potential bug fixes.

File: sample-client-app/sample-app-with-shared-lib/README.md [hunk 1]
```diff
@@ -33,4 +33,26 @@ How to build and run this sample application
    ```
    java -jar ./build/libs/sample-with-shared-lib-1.0.jar
    ```
-Note: The library jar will be created under "build/libs" directory.
+How to build and run this sample application as a Spring Boot executable jar
+------------
+ * Go to the root directory of this sample application
+   ```
+   g11n-java-client/sample/-client-app/sample-app-with-shared-lib
```
Comment (huihuiw01) [inline sample-client-app/sample-app-with-shared-lib/README.md:40]: Typo: sample/-client-app -> sample-client-app

**Review by surite — DISMISSED** [review]

File: sample-client-app/sample-app-with-shared-lib/build-spring-boot.gradle [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+sourceCompatibility = '1.8'
+archivesBaseName = 'sample-spring-boot-with-shared-lib'
+
+repositories {
+	mavenCentral()
+}
+
+dependencies {
+	implementation files('lib/sample-shared-library-1.0.jar')
+	implementation files('lib/vip4java-0.1.0.jar')
```
Comment (huihuiw01) [inline sample-client-app/sample-app-with-shared-lib/build-spring-boot.gradle:22]: Why not download it from maven repo?

File: sample-client-app/sample-app-with-shared-lib/build.gradle [hunk 3]
```diff
@@ -4,7 +4,7 @@
  */
 apply plugin: 'eclipse'
 apply plugin: 'java'
-project.group = 'com.vmware.sample'
+project.group = 'com.vmware.vipclient'
```
Comment (huihuiw01) [inline sample-client-app/sample-app-with-shared-lib/build.gradle:7]: Is it better set the group as 'com.vmware.singleton' ?

File: sample-client-app/sample-app-with-shared-lib/build-spring-boot.gradle (hunk 2 above)
Comment (jessiejuachon) [reply sample-client-app/sample-app-with-shared-lib/build-spring-boot.gradle:22]: We can take a look at this later in a separate PR as this has been like this before this PR.

File: sample-client-app/sample-app-with-shared-lib/build.gradle (hunk 3 above)
Comment (jessiejuachon) [reply sample-client-app/sample-app-with-shared-lib/build.gradle]: Actually, project.gorup does not matter for this as we are not publishing this artifact anywhere. To minimize the changes, I am reverting this line to the original.

File: sample-client-app/sample-app-with-shared-lib/build-spring-boot.gradle [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * Copyright 2019-2020 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+plugins {
+	id 'org.springframework.boot' version '2.3.3.RELEASE'
+	id 'io.spring.dependency-management' version '1.0.8.RELEASE'
+	id 'java'
+}
+
+group = 'com.vmware.vipclient'
```
Comment (huihuiw01) [inline sample-client-app/sample-app-with-shared-lib/build-spring-boot.gradle:11]: Since you revert the change for group, so this group should be consistent with others.

**Review by huihuiw01 — APPROVED** [review]

File: sample-client-app/sample-app-with-shared-lib/README.md [hunk 5]
```diff
@@ -7,7 +7,7 @@ Prerequisites
 ------------
  * Build the Singleton Java client library by following the instructions in [here](https://github.com/vmware/singleton/blob/g11n-java-client/README.md).
 
-How to build and run this sample application
+How to build and run this sample application as an executable jar
 ------------
  * Go to the root directory of this sample application
```
Comment (Xiaochao8) [inline sample-client-app/sample-app-with-shared-lib/README.md:12]: sample/-client-app -> sample-client-app

File: sample-client-app/sample-app-with-shared-lib/README.md [hunk 6]
```diff
@@ -33,4 +33,26 @@ How to build and run this sample application
    ```
    java -jar ./build/libs/sample-with-shared-lib-1.0.jar
    ```
-Note: The library jar will be created under "build/libs" directory.
+How to build and run this sample application as a Spring Boot executable jar
```
Comment (Xiaochao8) [inline sample-client-app/sample-app-with-shared-lib/README.md:36]: Need a blank line before subject line
```suggestion

How to build and run this sample application as a Spring Boot executable jar
```

File: sample-client-app/sample-app-with-shared-lib/src/main/java/com/vmware/vipclient/sample/springboot/Application.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+@SpringBootApplication
+public class Application {
+
+	public static void main(String[] args) {
+		SpringApplication.run(Application.class, args);
+	}
+
+	@Bean
+	public CommandLineRunner commandLineRunner(ApplicationContext ctx) {
```
Comment (Xiaochao8) [inline sample-client-app/sample-app-with-shared-lib/src/main/java/com/vmware/vipclient/sample/springboot/Application.java:24]: Why need a CommandLineRunner?

Comment (jessiejuachon) [reply sample-client-app/sample-app-with-shared-lib/src/main/java/com/vmware/vipclient/sample/springboot/Application.java:24]: It's just a way in Spring Boot to automatically run some logic after the application context is loaded. There are many ways to do it in Spring (https://reflectoring.io/spring-boot-execute-on-startup/), but we only need one here to run Main.main automatically which triggers all the sample code. Do you prefer any?

**Review by Xiaochao8 — APPROVED** [review]

**Review by huihuiw01 — APPROVED** [review]

**Comment by jessiejuachon** [comment]
@lyiyu66 , FYI, this has been merged.
